src/main/java/graphql/execution/directives/DirectivesResolver.java

```diff
@@ -26,17 +26,17 @@ public class DirectivesResolver {
     public DirectivesResolver() {
     }
 
-    public Map<String, List<GraphQLDirective>> resolveDirectives(List<Directive> directives, GraphQLSchema schema, Map<String, Object> variables, GraphQLContext graphQLContext, Locale locale) {
+    public BiMap<GraphQLDirective, Directive> resolveDirectives(List<Directive> directives, GraphQLSchema schema, Map<String, Object> variables, GraphQLContext graphQLContext, Locale locale) {
```

Comment:
The only use of this internal class is QueryDirectivesImpl - so we can change this with impunity

Reply:
Map<String, List<GraphQLDirective>> was a pointless modelling - BiMap<GraphQLDirective, Directive is what we need